src/ApplePay/Decoding/OpenSSL/OpenSslService.php

```diff
@@ -114,7 +97,27 @@ public function deriveKey($privateKeyFilePath, $publicKeyFilePath) {
             throw new \RuntimeException("Unexpected empty result");
         }
 
-        return base64_decode($execOutput[0]);
+        return $execOutput;
     }
 
+    private function runCommand($command)
+    {
+        $descriptorspec = [
+            1 => ["pipe", "w"],
+            2 => ["file", "/dev/null", "a"]
+        ];
+
+        $process = proc_open($command, $descriptorspec, $pipes);
+
+        if (!is_resource($process)) {
+            throw new \RuntimeException("Unable to invoke openssl");
+        }
+
+        $execOutput = stream_get_contents($pipes[1]);
+        fclose($pipes[1]);
+
+        $execStatus = proc_close($process);
+
+        return [$execStatus, $execOutput];
```

Comment:
Since this is called from multiple places, would be better to have this method return an object with command result, instead of array. 
It only needs to have two methods for getting the code and the output. You can pass the values in constructor.